### PR TITLE
Dependabot to update go deps

### PR DESCRIPTION
**Pull request by dimityrmirchev**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```feature user

```

### CODE REVIEW

File: .github/dependabot.yaml [hunk 1]
```diff
@@ -1,5 +1,10 @@
 version: 2
 updates:
+- package-ecosystem: gomod
+  directory: /
+  schedule:
+    interval: daily
+  open-pull-requests-limit: 5
```
[the remark this answers is not in the document]
Comment (dimityrmirchev) [reply .github/dependabot.yaml:7]: I wanted to see what other dependencies would be updated by the dependabot if we do not limit it to just g/g updates. As per https://github.com/gardener/gardener-extension-shoot-oidc-service/pull/141#issuecomment-1921200190 I think that such security updates will be skipped if g/g is explicitly set as dependency name.

[the remark this answers is not in the document]
Comment (vpnachev) [reply .github/dependabot.yaml:7]: Ok, let's give it a try.

**Review by vpnachev — APPROVED** [review]
/lgtm

**Pull request merged**